src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Shell/VSShellServices.cs

```diff
@@ -11,65 +12,54 @@ namespace Microsoft.VisualStudio.Shell;
 [AppliesTo(ProjectCapabilities.AlwaysApplicable)]
 internal class VSShellServices : IVsShellServices
 {
-
-    private readonly AsyncLazy<(bool IsCommandLineMode, bool IsPopulateSolutionCacheMode)> _initialization;
+    private static readonly object s_lockObject = new();
+    private static Task<bool>? s_isCommandLine;
+    private static Task<bool>? s_isPopulateCache;
```

Comment:
This particular service is global in scope (per contract attribute on `IVsShellServices`) so these don't have to be static/locked, nor do they have to be nullable.

In your testing, did you see this class be instantiated more than once per VS session? If so, that would be a bug elsewhere.

Reply:
Ah, no, just a misunderstanding of the scope of this service on my part.